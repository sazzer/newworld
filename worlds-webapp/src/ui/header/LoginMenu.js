// @flow

import React from 'react';
import {Interpolate} from 'react-i18next';
import {Menu} from 'semantic-ui-react'
import {connectStore} from "redux-box";
import type {AuthModule} from '../../auth';
import {module as authModule} from '../../auth';

/** The flow type representing the props for the Login Menu */
type LoginMenuProps = {
    onClickLogin: () => void,
};

/**
 * React component to represent the area of the header bar that is clicked to log in
 */
export function LoginMenu(props: LoginMenuProps) {
    return (
        <Menu.Item onClick={props.onClickLogin}>
            <Interpolate i18nKey="page.login" />
        </Menu.Item>
    );
}

/**
 * Wrapper around the Login Menu that connects it to Redux
 */
export function ConnectedLoginMenu({auth}: {auth: AuthModule}) {
    return (
        <LoginMenu onClickLogin={auth.startAuth} />
    )
}

export default connectStore({auth: authModule})(ConnectedLoginMenu);
