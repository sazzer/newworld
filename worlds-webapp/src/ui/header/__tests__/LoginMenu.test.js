// @flow

import React from 'react';
import {shallow} from 'enzyme';
import {LoginMenu} from '../LoginMenu';

/** Create the component to test */
function setup() {
    const onClickLogin = jest.fn();

    return {
        onClickLogin,
        element: shallow(<LoginMenu onClickLogin={onClickLogin()} />)
    };
}

describe('LoginMenu', () => {
    it('Renders correctly', () => {
        const { element } = setup();

        expect(element).toMatchSnapshot();
    });

    it('Reacts to being clicked', () => {
        const { onClickLogin, element } = setup();

        element.simulate('click');

        expect(onClickLogin).toHaveBeenCalledTimes(1);
    });
});
