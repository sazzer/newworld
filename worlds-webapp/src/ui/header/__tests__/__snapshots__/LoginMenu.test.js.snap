// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginMenu Renders correctly 1`] = `
<MenuItem
  data-test="loginMenu"
>
  <Interpolate
    className=""
    i18nKey="page.login"
  />
</MenuItem>
`;
